src/AggregateFunctions/AggregateFunctionTopK.cpp

```diff
@@ -279,7 +279,7 @@ class AggregateFunctionTopKGeneric final
                 size,
                 getName(),
                 TOP_K_MAX_SIZE);
-        set.resize(size);
+        set.resize(std::max(size_t(reserved), size));
```

Comment:
Seems to be too much, tests fail because of OOMs.
Maybe `set.resize(std::min(size + 1, reserved));`?

Reply:
Ah yeah, that's a better idea than my patch in general. Changed accordingly, let's see how it goes this time. :)